File: How_To_Program_Exercises/C++How_To_Program_8/Chapter3/EmployeeClass/EmployeeTest.cpp
//
//	Employee Test 
//

#include <iostream>
#include <string>
#include "Employee.h"

int main ()
{
	std::cout <<"\nCreating first instance of class Employee\n-------------------------------------------\n";
	Employee bojack("Bojack", "Horseman", 2500);
	
	std::cout <<"First Name: " << bojack.getFirstName()
			  <<"\nLast Name: " << bojack.getLastName()
			  <<"\nMonthly Salary: $" << bojack.getMonthlySalary()
			  <<"\nYearly Salary: $" << bojack.getMonthlySalary() * 12 << '\n';
	
	
	std::cout <<"\nCreating second instance of class Employee\n---------------------------------------------\n";
	Employee morty("Morty", "Smith", 4500);
	
	std::cout <<"First Name: " << morty.getFirstName()
			  <<"\nLast Name: " << morty.getLastName()
			  <<"\nMonthly Salary: $" << morty.getMonthlySalary()
			  <<"\nYearly Salary: $" << morty.getMonthlySalary() * 12 << '\n';
			  
	//Salary Increase
	bojack.setMonthlySalary(bojack.getMonthlySalary() * 110 / 100);
	morty.setMonthlySalary(morty.getMonthlySalary() * 110 / 100);
			  
	std::cout <<"\n\nAfter 10% raise\n------------------\n";
	std::cout <<"Bojack's yearly salary: $" << bojack.getMonthlySalary() * 12
			  <<"\nMorty's yearly salary: $" << morty.getMonthlySalary() * 12
			  <<std::endl;
	
	
}

File: How_To_Program_Exercises/JavaHowToProgram9/Chapter3/EmployeeClass/EmployeeTest.java
//
//	Employee Test Class
//

class EmployeeTest
{
	public static void main (String[] args)
	{
		System.out.println("\nCreating first instance of class Employee\n-------------------------------------------\n");
		Employee bojack = new Employee ("Bojack", "Horseman", 2500.00);
		
		System.out.printf("First Name: %s\nLast Name: %s\nMonthly Salary: $%.2f\nYearly Salary: $%.2f\n",
						  bojack.getFirstName(), bojack.getLastName(), bojack.getMonthlySalary(), bojack.getMonthlySalary() * 12);
		
		System.out.println("\n\n\nCreating second instance of class Employee\n---------------------------------------------\n");
		Employee morty = new Employee ("Morty", "Smith", 4000.00);
		
		System.out.printf("First Name: %s\nLast Name: %s\nMonthly Salary: $%.2f\nYearly Salary: $%.2f\n",
						  morty.getFirstName(), morty.getLastName(), morty.getMonthlySalary(), morty.getMonthlySalary() * 12);
	
		//Salary Increase
		morty.setMonthlySalary(morty.getMonthlySalary() * 1.1);
		bojack.setMonthlySalary(bojack.getMonthlySalary() * 1.1);
		
		System.out.println("\n\nAfter 10% raise\n-----------------\n");
		System.out.printf("Bojack's yearly salary: $%.2f\nMorty's yearly salary $%.2f\n",
						  bojack.getMonthlySalary() * 12, morty.getMonthlySalary() * 12);
	}
}